js/coinfloor.js

```diff
@@ -290,21 +290,20 @@ module.exports = class coinfloor extends Exchange {
         }
         if (this.safeString (item, 'type') === '2') {
             // it's a trade so let make 2 entries
-            const orderId = this.safeString (item, 'order_id');
```

Comment:
Why  discard the order_id as the reference id?

Reply:
Because a ledger entry of type `trade` should have `referenceId` equal to the related trade id, not to the order id. We don't have a ledger entry type for `order`.

Reply:
Oh I see, so you use the trade id as the reference id, and discard the id altogether (for trades).  You seem to have discarded the id for transfers though, which confused me.  Was this intentional ?

Reply:
> You seem to have discarded the id for transfers though, which confused me. Was this intentional ?

I don't think i missed it, I moved them both to `referenceId`. The logic within the ledger entry structure is as follows:

```JavaScript
{
    id: 'abc123', // this is the ledger entry id, must be unique
    ...
    type: 'trade', // or 'transfer', or 'transaction', ...
    ...
    referenceId: 'def456', // id of the underlying `type` structure related to this ledger entry
    ...
}
```

Let me know if I'm still missing something.

Reply:
Ok, I understand.  Just because an id is unique, that doesn't mean its suitable for a ledger id.  You're assuming that if there is a numeric id, and it ISN'T increasing atomically, then it's not a ledger id. 

What isn't clear, is that some exchanges don't use numeric id's, so you have no way to determine if or not they are suitable for a ledger id or not, right?

Reply:
> What isn't clear, is that some exchanges don't use numeric id's, so you have no way to determine if or not they are suitable for a ledger id or not, right?

Exactly. Sometimes exchanges name those fields in a way that allows to determine what is what. In this case, however, those ids really designate the ids of the related structures, not the ids of the ledger entries.